packages/docusaurus/src/commands/deploy.ts

```diff
@@ -63,8 +63,22 @@ This behavior can have SEO impacts and create relative link issues.
     throw new Error('Git not installed or on the PATH!');
   }
 
+  const currentRepoUrl = shell
+    .exec('git config --get remote.origin.url')
+    .stdout.trim();
+
+  const repoUrlUseSSH =
+    currentRepoUrl.match(/^ssh:\/\//) !== null ||
+    currentRepoUrl.match(/^([\w-]+@)?[\w.-]+:[\w./_-]+(\.git)?/) !== null;
```

Comment:
Where do these regexp come from? If you found there somewhere I'd appreciate a link to an immutable URL as comment.

Can you add test cases for those?

Reply:
This seems... intuitive to me🧐

It's either `ssh://***` or `git@org:repo.git`

Reply:
A regexp is a less readable shorthand for real code and must be tested in the same way. Similarly an if/else is intuitive until there's a bug in it 🤪